db/trie/merklepatriciatree/twolayertrie.go

```diff
@@ -70,81 +84,109 @@ func (tlt *twoLayerTrie) Start(ctx context.Context) error {
 	layerOne, err := New(
 		KVStoreOption(tlt.kvStore),
 		RootHashOption(rootHash),
+		AsyncOption(),
```

Comment:
the idea is to enable async, so all the nodes hash/serialize will be done in one stroke at the time of Stop() and write into DB, is that right?

can we run TestPressure() in trie_test.go, and measure the improvement?

and 2 more questions:
1. why we need 2-layer trie, what is the first layer used for?
2. how do we keep the Root hash same as before?

Reply:
also where is this "cancel out" done in your change?
i mean, if there are 2 DB op on the same key, the later 2nd value overwrites the 1st value

Reply:
> the idea is to enable async, so all the nodes hash/serialize will be done in one stroke at the time of Stop() and write into DB, is that right?
> 
> can we run TestPressure() in trie_test.go, and measure the improvement?
> 
> and 2 more questions:
> 
> 1. why we need 2-layer trie, what is the first layer used for?
> 2. how do we keep the Root hash same as before?

it is for historical trie, to store different namespaces. The first layer is for namespace, the second layer is for the values in the namespace.

Reply:
we talked about earlier, can use a map to cache the namespaces, this can skip the first layer look-up and gain some speed-up

do you think can add into this PR, or a new PR?

Reply:
Since you have some new idea to improve it, would you mind implement this part?

Reply:
ok

Reply:
2 unresolved question:

1. how to keep the Root hash (of layer-2 trie) same as before?

2. can you point out, where is this "cancel out" done in your change?
i mean, if there are 2 DB op on the same key, the later 2nd value overwrites the 1st value